File: Learnify.Domain/Entities/Module.cs
using System.Text.Json.Serialization;

namespace Learnify.Domain.Entities
{
    public class Module : BaseEntity
    {
        public string Title { get; set; } = string.Empty;
        public Guid CourseId { get; set; }
        [JsonIgnore] // Add this attribute
        public Course? Course { get; set; }
        public int Order {  get; set; }
        public virtual ICollection<Lesson>? Lessons { get; set; }
    }
}

File: Learnify.UI/src/models/entities/Module.js
export class Module {
    constructor(
        id,
        title,
        courseId,
        course = null,
        order,
        lessons = [],
    ) {
        this.id = id;
        this.title = title;
        this.courseId = courseId;
        this.course = course;
        this.order = order;
        this.lessons = lessons;
        
    }
}